.github/workflows/package-release.yml

```diff
@@ -0,0 +1,113 @@
+name: "Package Release"
+run-name: "Package Release ${{ github.ref_name }}"
+
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  build-pecl:
+    name: "Create PECL package"
+    runs-on: "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v4"
+        with:
+          # Manually specify a ref. When actions/checkout is run for a tag without a ref, it looks up the underlying
+          # commit and specifically fetches this to the refs/tags/<tag> ref, which denies us access to the tag message
+          ref: ${{ github.ref }}
+          fetch-depth: 2
+          submodules: true
+
+      - name: "Build Driver"
+        uses: ./.github/workflows/linux/build
+        with:
+          version: "8.3"
+
+      - name: "Write changelog file for packaging"
+        run: git tag -l ${{ github.ref_name }} --format='%(contents)' > changelog
+
+      - name: "Build package.xml"
+        run: "make package.xml"
+
+      - name: "Build release archive"
+        run: "make package"
+
+      # PECL always uses the version for the package name.
+      # Read it from the version file and store in env to use when uploading artifacts
+      - name: "Read current package version"
+        run: echo "PACKAGE_VERSION=$(./bin/update-release-version.php version)" >> "$GITHUB_ENV"
+
+      # Currently fails with: Cannot install, php_dir for channel "pecl.php.net" is not writeable by the current user
+      # - name: "Install release archive to verify correctness"
+      #  run: pecl install mongodb-${{ env.PACKAGE_VERSION }}.tgz
+
+      - name: "Upload artifact"
+        uses: actions/upload-artifact@v4
+        with:
+          name: mongodb-${{ env.PACKAGE_VERSION }}.tgz
+          path: mongodb-${{ env.PACKAGE_VERSION }}.tgz
+          retention-days: 3
+
+      - name: "Upload release artifact"
+        run: gh release upload ${{ github.ref_name }} mongodb-${{ env.PACKAGE_VERSION }}.tgz
+        # Until we've changed the process to always have a release draft, ignore errors during upload
```

Comment:
The comment was left over from an initial prototype where I hadn't created the release yet, so it's definitely obsolete.

However, I'm leaving `continue-on-error` in for the time being, as the process right now supports building packages for tags that don't have a release associated with them (think manually pushing a tag). Once we've confirmed that all our release types are appropriately covered, we can remove this and let the build fail if it can't find a release for the tag being built.